components/selection/selection-observer-mixin.js

```diff
@@ -16,29 +16,29 @@ export const SelectionObserverMixin = superclass => class extends superclass {
 			 * @type {object}
 			 */
 			selectionInfo: { type: Object },
-			_provider: { type: Object, attribute: false }
+			_provider: { state: true }
 		};
 	}
 
 	constructor() {
 		super();
 		this.selectionInfo = new SelectionInfo();
-		this._provider = null;
+		this.__provider = null;
 	}
 
 	connectedCallback() {
 		super.connectedCallback();
 		// delay subscription otherwise import/upgrade order can cause selection mixin to miss event
 		requestAnimationFrame(() => {
-			if (this.selectionFor) return;
+			if (this.selectionFor || this._provider) return;
```

Comment:
It seems like there is an edge case here where if the observing element is disconnected from the DOM and then reconnected elsewhere in the DOM, it wouldn't get a new provider assigned. 

For the event code path immediately below, that could be mitigated by setting the `_provide = null` in the `disconnectedCallback` - I am almost a little surprised it isn't already doing that. We must be assuming this edge case isn't worth dealing with...? I think setting it to `null` in the `disconnectedCallback` complicates the case where the `_provider` is assigned directly though.

Reply:
I have a fix for this, but it has its own edge case in the default event-driven behaviour that may cause an issue.

@dbatiste I've got a rewrite in the works using the full event-driven approach we talked about on Friday, rather than using `._provider`, I think it may end up cleaner.